astropy/io/ascii/tests/test_read.py

```diff
@@ -1493,3 +1495,43 @@ def test_kwargs_dict_guess(enable):
     for k in get_read_trace():
         if not k.get('status', 'Disabled').startswith('Disabled'):
             assert k.get('kwargs').get('fast_reader').get('enable') is enable
+
+
+@pytest.mark.parametrize('rdb', [False, True])
+@pytest.mark.parametrize('fast_reader', [False, 'force'])
+def test_deduplicate_names_basic(rdb, fast_reader):
+    """Test that duplicate column names are successfully de-duplicated for the
+    basic format.
+    """
+    lines = ['a a_2 a_1 a a']
+    if rdb:
+        lines += ['N N N N N']
+    lines += ['1 2 3 4 5', '10 20 30 40 50']
+
+    if rdb:
+        lines = ['\t'.join(line.split()) for line in lines]
+
+    dat = ascii.read(lines, fast_reader=fast_reader)
+    assert dat.colnames == ['a', 'a_2', 'a_1', 'a_3', 'a_4']
+
+    with ExitStack() as stack:
+        # include_names doesn't work for the fast_reader and RDB (even for unique
+        # colnames), so expect an exception in this case.
+        if rdb is True and fast_reader == 'force':
+            stack.enter_context(pytest.raises(AssertionError))
+
+        dat = ascii.read(lines, fast_reader=fast_reader, include_names=['a', 'a_2', 'a_3'])
```

Comment:
Why is this all under "with ExitStack()"? This seems to be a separate test.

Reply:
Sorry, only spotted now that `include_names` had already been identified as the culprit in the earlier commits. Still, the current version gives the misleading impression that it is primarily a problem of the duplicate colnames.